sentinel-core/src/main/java/com/alibaba/csp/sentinel/util/AppNameUtil.java

```diff
@@ -57,7 +55,9 @@ private AppNameUtil() {
 
     static {
         resolveAppName();
-        RecordLog.info("App name resolved: " + appName);
+        //here use RecordLog will lead to class-init circle-dependency problem
+        //RecordLog.info("App name resolved: " + appName);
+        System.out.println("App name resolved: " + appName);
```

Comment:
Here we might need to print the `appName` to the log as it's significant information. That's why we had separated logging related properties from the `SentinelConfig`. Are there any solutions?

Reply:
> Here we might need to print the `appName` to the log as it's significant information. That's why we had separated logging related properties from the `SentinelConfig`. Are there any solutions?

If the configuration is isolated, my idea is to add two new loader classes. `SentinelConfigLocator` is responsible for loading the Sentinel Core configuration, and `LogConfigLocator` is responsible for loading the Log configuration. `SentinelConfig` reads the configuration from `SentinelConfigLocator`, and `LogBase` reads the configuration from `LogConfigLocator`. This way they are completely independent, and the problem of class initialization interdependence is also solved. What needs to be considered now is whether the Log configuration and Sentinel Core are placed on the same configuration file? If so, `SentinelConfigLocator`, `LogConfigLocator` will read the same file. Is this ok?